File: python/calculator.py
import os

option = 0
menu = ["Add", "Substract", "Multiply", "Divide", "Exit"]
first = 0
second = 0

def add(a, b):
    return a + b

def substract(a, b):
    return a - b

def multiply(a, b):
    return a * b

def divide (a, b):
    return a / b

if __name__ == "__main__":
    while option != len(menu) + 1:
        os.system("cls")
        if second == 0:
            pass
        elif option == 1:
            print(f"{first} + {second} = {add(first, second)}\n")
        elif option == 2:
            print(f"{first} - {second} = {substract(first, second)}\n")
        elif option == 3:
            print(f"{first} * {second} = {multiply(first, second)}\n")
        elif option == 4:
            print(f"{first} / {second} = {divide(first, second)}\n")
        else:
            print("Invalid option")
        for i, e in enumerate(menu):
            print(f"{(i + 1)}. {e}", end="  ")

        option = int(input("\nSelect an option:\n"))
        first = int(input("Select a 1st number:\n"))
        match option:
            case 1:
                second = int(input(f"Select a 2nd one: \n{first} + "))
            case 2:
                second = int(input(f"Select a 2nd one: \n{first} - "))
            case 3:
                second = int(input(f"Select a 2nd one: \n{first} * "))
            case 4:
                second = int(input(f"Select a 2nd one: \n{first} / "))
            case _:
                second = 0

        if option == len(menu) + 1:
            print("Bye!")
        elif option == 1:
            print(f"{first} + {second} = {add(first, second)}")
        elif option == 2:
            print(f"{first} - {second} = {substract(first, second)}")
        elif option == 3:
            print(f"{first} * {second} = {multiply(first, second)}")
        elif option == 4:
            print(f"{first} / {second} = {divide(first, second)}")
        else:
            print("Invalid option")

File: dart/calculator.dart
// No copilot here, I'm alone in the cockpit.
import 'dart:io';

var menu = ["Addition", "Substraction", "Multiplication", "Division", "Exit"];
int option = 0;
int result = 0;
int first = 0;
int second = 0;
int i = 0;
int running = 0;

int add(a, b) {
    return a + b;
}
int sub(a, b) {
    return a - b;
}
int multiply(a, b) {
    return a * b;
}
int divide(a, b) {
    return a ~/ b;
}

void main() {
    while (running == 0) {
        print("Select an option: ");
        for (i = 0; i < 5; i++) {
            stdout.write("${i + 1}. ${menu[i]}  ");
        }
        option = int.parse(stdin.readLineSync()!);

        if (option == menu.length) {
            running = 1;
            break;
        }
        else if (option < 0 || option > menu.length) {
            print("Invalid input !");
            main();
        }
        print("First number: ");
        first = int.parse(stdin.readLineSync()!);
        print("Second number: ");
        second = int.parse(stdin.readLineSync()!);
        if (option == 1) {
            result = add(first, second);
            print("${first} + ${second} = ${result}");
        }
        else if (option == 2) {
            result = sub(first, second);
            print("${first} - ${second} = ${result}");
        }
        else if (option == 3) {
            result = multiply(first, second);
            print("${first} * ${second} = ${result}");
        }
        else if (option == 4) {
            result = divide(first, second);
            print("${first} / ${second} = ${result}");
        }
    }
    
}


/*
Ce que j'ai appris de Dart :

    1. import 'dart:io'; = cette lib permet d'importer dans ce code des fonctions dérivée d'output "stdin.readLineSync()"
    2. void = définit des fonctions non-typées qui n'ont pas de paramètres, donc ne retournent rien
    3. J'ai du faire un "état" "running" car je n'ai pas su faire true/false dans ma while loop

*/
